[0.5.0-UT] test_fused_attention ut fix

gulsumgudukbay opened the pull request:
decreased seq_len in tests/pallas/gpu_ops_test.py::FusedAttentionTes::test_fused_attention tests

JehandadKhan commented:
This change as-is cannot be upstreamed! Please add guards for ROCm and also rationale for why we have to reduce it!

mrodden commented:
Can you add a blurb in the commit message or in a comment inline as to why this is needed and how it makes the tests pass?

gulsumgudukbay commented:
> Can you add a blurb in the commit message or in a comment inline as to why this is needed and how it makes the tests pass?

I added this in the commit message.

gulsumgudukbay commented:
> This change as-is cannot be upstreamed! Please add guards for ROCm and also rationale for why we have to reduce it!

added the guards for ROCm and rationale. the seq_len is assigned conditionally to lower dimension for both tests. I also added the rationale as a comment block before each test.
